cloud/azure/keyvault/detectors-keyvault.tf

```diff
@@ -0,0 +1,76 @@
+resource "signalfx_detector" "heartbeat" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Azure keyvault heartbeat"
+
+  program_text = <<-EOF
+		from signalfx.detectors.not_reporting import not_reporting
+		signal = data('Availability', filter=filter('resource_type', 'Microsoft.KeyVault/vaults') and filter('primary_aggregation_type', 'true') and ${module.filter-tags.filter_custom}).publish('signal')
```

Comment:
Based on Azure documentation in https://docs.microsoft.com/en-us/azure/key-vault/general/alert
Vault Availability - This metric should always be at 100% this is an important metric to monitor, since it can quickly show you if your key vault experienced an outage.